bootstrap/kubeadm/types/upstreamv1beta3/types.go

```diff
@@ -225,6 +225,13 @@ type NodeRegistrationOptions struct {
 	// IgnorePreflightErrors provides a slice of pre-flight errors to be ignored when the current node is registered.
 	// +optional
 	IgnorePreflightErrors []string `json:"ignorePreflightErrors,omitempty"`
+
+	// ImagePullPolicy specifies the policy for image pulling
+	// during kubeadm "init" and "join" operations. The value of
+	// this field must be one of "Always", "IfNotPresent" or
+	// "Never". Defaults to "IfNotPresent".
+	// +optional
+	ImagePullPolicy string `json:"imagePullPolicy,omitempty"`
```

Comment:
Took me a bit to figure it out / remember.

We have a custom marshal func for NodeRegistrationOptions. You have to add ImagePullPolicy there too:

![image](https://user-images.githubusercontent.com/4662360/214657391-e432f25b-cc48-4c0c-90c3-cc097a030f20.png)


Otherwise roughly something like this happens:
* kubectl sends the ImagePullPolicy Always
* the KCP mutate webhook goes through marshal / unmarshal JSON and drops the ImagePullPolicy
* OpenAPI defautling defaults the field that is unset after the mutating webhook to IfNotPresent

Reply:
Thanks! It worked.